### PR TITLE
Add TX, TN, and TG days above and below

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR works towards adrressing #714 
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 

This adds four "new" indicators and indices (`tx_days_below`, `tg_days_above`, `tg_days_below`, `tn_days_above`) so that we don't need to leverage an oddly-named indicator to perform a general type of indices calculation.

* **Does this PR introduce a breaking change?** 

No, but we currently have a `tropical_nights` that is programmatically identical to `tn_days_above`. It should eventually be replaced.

* **Other information**:

### CODE REVIEW

File: xclim/data/fr.json [hunk 1]
```diff
@@ -192,20 +192,48 @@
...
+    "title": "Nombre de jours avec tmoy au-dessus d'un certain seuil",
+    "comment": "",
+    "abstract": "Nombre de jours où la moyenne de température journalière est au-dessus d'un seuil."
+  },
+  "TG_DAYS_BELOW": {
+    "long_name": "Nombre de jours avec Tmoy < {thresh}",
+    "description": "Nombre {freq:m} de jours où la moyenne de température journalière est sous {thresh}",
+    "title": "Nombre de jours avec tmoy sous un certain seuil",
+    "comment": "",
+    "abstract": "Nombre de jours où la moyenne de température journalière est sous un seuil."
```
Comment (aulemahal) [inline xclim/data/fr.json:221]:
```suggestion
    "description": "Nombre {freq:m} de jours où la température journalière moyenne est au-dessus de {thresh}",
    "title": "Nombre de jours avec tmoy au-dessus d'un certain seuil",
    "comment": "",
    "abstract": "Nombre de jours où la température journalière moyenne est au-dessus d'un seuil."
  },
  "TG_DAYS_BELOW": {
    "long_name": "Nombre de jours avec Tmoy < {thresh}",
    "description": "Nombre {freq:m} de jours où la température journalière moyenne est sous {thresh}",
    "title": "Nombre de jours avec tmoy sous un certain seuil",
    "comment": "",
    "abstract": "Nombre de jours où la température journalière moyenne est sous un seuil."
```

**Review by aulemahal — APPROVED** [review]
Thanks! First step of a needed clean up of xclim!

I agree with keeping the `tropical_nights` indicator to reduce breaking changes, but I would vote to remove the indice itself.

**Review by tlogan2000 — CHANGES_REQUESTED** [review]
Looks good. I think the new indicators will require french translations though in the locals json..

**Comment by Zeitsperre** [comment]
> Looks good. I think the new indicators will require french translations though in the locals json..

I thought I already addressed that? https://github.com/Ouranosinc/xclim/pull/715/files#diff-4dfa328c013bf54d291f1c211de4838b0cb813f39ce930578b059ca1daace275

**Comment by tlogan2000** [comment]
> > Looks good. I think the new indicators will require french translations though in the locals json..
> 
> I thought I already addressed that? https://github.com/Ouranosinc/xclim/pull/715/files#diff-4dfa328c013bf54d291f1c211de4838b0cb813f39ce930578b059ca1daace275

My mistake missed it

**Review by tlogan2000 — APPROVED** [review]

**Comment by Zeitsperre** [comment]
Will merge as soon as docs and (most) tests pass.

**Pull request merged**